waku/v2/utils/time.nim

```diff
@@ -9,15 +9,15 @@ type Timestamp* = int64
 const TIMESTAMP_TABLE_TYPE* = "INTEGER"
 
 proc getNanosecondTime*[T](timeInSeconds: T): Timestamp = 
-  var ns = Timestamp(timeInSeconds*1000_000_000)
+  var ns = Timestamp(timeInSeconds.int64 * 1000_000_000.int64)
```

Comment:
Right. I don't have a strong preference here, but I think I like the explicit conversions to `int64` better: we're unlikely to change `Timestamp` and, even then, it might make sense to revisit our conversions when we do to ensure that we don't do anything unexpected. Also I find `Timestamp(1000_000_000)` a bit more obscure than a multiplier of `1000_000_000.int64`. Wdyt?

Reply:
> To keep consistency with the Timestamp type (that we might later change to a new type different than int64), would it be better to convert both operands to Timestamp type instead?

@s1fr0  If we use this approach, then there can be cases that this proc will error out i.e., when the type of `timeInSeconds` has a bit size smaller than 64 e.g., `int8`, the reason is that `1000_000_000` cannot  fit into `int8` (the minimum int size for it is 64`)

@jm-clius Though there is still a problem with the current version, and that is if `timeInSeconds` happens to have a value that does not fit in `int64` e.g., `uint64.high`, then this proc will emit an exception. The safest would be to limit the input type `T` to only signed values i.e., `int8,16,32,64`?

Reply:
Why the first comment is the case? if e.g. `timeInSeconds` is an int8, `Timestamp(timeInSeconds)` will cast it to int64 and then 1000_000_000 should hopefully cast to int64 as well.

I clearly agree to your second comment.

Reply:
> Why the first comment is the case? if e.g. timeInSeconds is an int8, Timestamp(timeInSeconds) will cast it to int64 and then 1000_000_000 should hopefully cast to int64 as well.

@s1fr0 
Oh, I meant:
> there can be cases that this proc will error out i.e., when the  **`TimeStamp`** type ~of timeInSeconds~ has a bit size smaller than 64 e.g., int8, the reason is that 1000_000_000 cannot fit into int8 (the minimum int size for it is 64`)